includes/class-amp-story-post-type.php

```diff
@@ -1368,4 +1371,48 @@ public static function remove_title_from_embed( $output, $post ) {
 
 		return preg_replace( '/<blockquote class="wp-embedded-content">.*?<\/blockquote>/', '', $output );
 	}
+
+	/**
+	 * Changes attributes of the AMP Story embed <iframe>.
+	 *
+	 * The embed typically appears in the editor in an <iframe>, and without an <iframe> on the front-end.
+	 * The height of the <iframe> isn't enough to display the full story, so this increases it.
+	 * Also, this removes attributes that AMP does not allow for the <amp-iframe>, like marginwidth.
+	 *
+	 * @param string  $output The embed output.
+	 * @param WP_Post $post The post for the embed.
+	 * @return string The filtered embed output.
+	 */
+	public static function change_embed_iframe_attributes( $output, $post ) {
+		if ( self::POST_TYPE_SLUG !== get_post_type( $post ) ) {
+			return $output;
+		}
+
+		preg_match( '/<iframe sandbox="allow-scripts" [^>]*>/', $output, $matches );
+		if ( ! $matches ) {
+			return $output;
+		}
+
+		$original_iframe = $matches[0];
+		$new_iframe      = preg_replace(
+			'/(?<=\sheight=")\w+(?=")/',
+			strval( ( self::STORY_LARGE_IMAGE_DIMENSION / 2 ) + 4 ),
+			$original_iframe
+		);
+
+		// Remove attributes that AMP doesn't allow for the <amp-iframe>.
+		if ( is_amp_endpoint() ) {
+			$new_iframe = preg_replace(
```

Comment:
@kienstra In which case would this replacement be used? If `amp-iframe` isn't used in the front-end and in the editor the `iframe` element is used (if I understood it correctly).

Reply:
@miina, good question. There were validation errors above the block for these before, but I can't reproduce them now.

Reply:
https://github.com/ampproject/amp-wp/pull/2245/commits/144e09b021911ca6f71e5690497e06310d4b071b removes the logic that removed these attributes.